R/zchunk_LB165.ag_water_R_C_Y_GLU_irr.R

```diff
@@ -0,0 +1,407 @@
+      # NOTE: dropping all country / GLU / crops with zero production; these won't have any water anyway
+      filter(Prod_t > 0) %>%
+      mutate(coef_m3kg = value / (Prod_t * CONV_T_KG)) ->
+      L165.ag_Water_ctry_MHcrop_GLU
+
+    # Clip extremely high values which are due to statistical discrepancies (original lines 84-106)
+    # The assumed maximum values are crop specific and based on analysis of the 2011 inventory data
+    # In this step, we're adding 2 m3/kg to every maximum observed value across all national averages.
+    # We want to use something a bit higher than the maximum of the national averages, because the
+    # sub-national regions' values will have a greater range.
+    #
+    # Still, this sort of cap is necessary because many of the water quantities, particularly in
+    # small regions, are clearly inconsistent with the underlying production data and produce
+    # extremely high water demand coefficients. In the case of Barley in Iraq it is clearly an
+    # error in the M+H gridded inventory, but for others it could be a simple mis-match in
+    # estimates of crop production by region, and share of irrigated production by region.
+    CTRY_GLU_MAX_ADDER <- 2
+    L165.Mekonnen_Hoekstra_Rep47_A2 %>%
+      group_by(FAO_crop) %>%
+      summarise(Blue = max(Blue + CTRY_GLU_MAX_ADDER),
+                Green = max(Green + CTRY_GLU_MAX_ADDER)) %>%
+      left_join(select(FAO_ag_items_PRODSTAT, item, MH_crop), by = c("FAO_crop" = "item")) ->
+      L165.MaxWaterCoefs_m3kg
+
+    # Change L165.ag_Water_ctry_MHcrop_GLU$coef_m3kg to be the minimum of itself and
+    # the corresponding (by crop and water_type) value in L165.MaxWaterCoefs_m3kg
+    L165.MaxWaterCoefs_m3kg %>%
+      gather(water_type, value_other, Blue, Green) %>%
+      mutate(water_type = tolower(water_type)) %>%
+      right_join(L165.ag_Water_ctry_MHcrop_GLU, by = c("MH_crop", "water_type")) %>%
+      mutate(coef_m3kg = pmin(coef_m3kg, value_other)) %>%
+      select(-value_other, -FAO_crop) ->
+      L165.ag_Water_ctry_MHcrop_GLU
+
+    # Work through calculation sequence to determine blue and green water coefficients for irrigated
+    # and rainfed production for gridded crops (original lines 108-122). The sequence has 5 steps:
+    #	(1) Calculate the total biophysical (green + blue) water quantities and coefficients for each country, GLU, and crop
+    #	(2) Match in the irrigated production and calculate the blue water coef for irrigated as min(blue water quantity / irr prod, total biophysical coef)
+    #	(3) Calculate the green water coef for irrigated crops as the total biophysical coef minus the blue water coef
+    #	(4) Calculate the total volume of green water on rainfed crops, as the total green water minus green water used by irrigated crops
+    #	(5) Calculate the green water coef for rainfed crops as rainfed green water volume divided by rainfed production
+    L165.ag_Water_ctry_MHcrop_GLU %>%
```

Comment:
It would be helpful to know which of the 5 sequence steps is being carried out at each point. It's easy to trace through with the comments, but it would probably save an external user time to have them explicitly labeled.